spec-0007/index.md

```diff
@@ -79,7 +80,86 @@ The following users will be affected:
 
 ### Code
 
-TODO: Add example `check_random_state` implementation.
+For example, SciPy may implement this with the following decorator to support both names.
+This is implemented using:
+1. A `check_random_state` function which normalizes the either old or new input to a `Generator` or `RandomState` object.
+   This function only gives a `FutureWarning` when users (probably) NumPy's global legacy `RandomState`.  As noted in point 1. above.
+2. A decorator to deal with the rename changes.  In future versions this will deprecate the old keyword.  At this time, it will ensure that the documentation and auto-completion only shows the new parameter name.
+   Delaying the deprecation ensures that downstream users can switch to `rng=` on all supported SciPy versions when the deprecation happens.
+
+```python
+_NoValue = object()  # singleton to indicate not explicitly passed
+
+
+def check_random_state(seed=_NoValue, rng=_NoValue):
+    if rng is not _NoValue and seed is not _NoValue:
+        raise TypeError("cannot pass both `rng=` and `random_state=` at the same time.")
+    if rng is not _NoValue:
+        return np.random.default_rng(rng)
+
+    if seed is _NoValue:
+        # If the user passed nothing, we have to reach into NumPy here:
+        # 1. If np.random.seed(None) was called (or never called), then we can
+        #    just use the default_rng (the result is random anyway).
+        # 2. If it was called, we must return the global random state object
+        #    and warn about future ignoring of seed!
+        if np.random.mtrand._rand._bit_generator._seed_seq is not None:
+            # The user did not seed, so no need to warn.
+            return np.random.default_rng()
+        warnings.warn(
+            "The NumPy global rng was seeded in call to np.random.seed() "
+            "in the future this function will ignore this seed and return "
+            "random values as if a new `np.random.default_rng()` was created.",
+            FutureWarning, stacklevel=5)
+        return np.random.mtrand._rand
+    if seed is None or seed is np.random:
+        return np.random.mtrand._rand
```

Comment:
np.random.Generator(np.random.mtrand._rand._bit_generator)

Reply:
Hmmm, right.  Although in some cases using a `Generator` will also generate different streams.  OTOH, if integers are used, maybe we should risk it (since that is the future stability anyway).

Maybe can still choose to do that, there could be some cases where a RandomState is lacking features.

And at least unless we worry a lot about stream stability, `rng=` could at least refuse `RandomState` so it gets phased out eventually.